MozillaFoundation/foundation.mozilla.org
# Added form submission thank you message on donate help page

# Description


Link to test page: https://foundation-s-11391-dona-arwuyu.herokuapp.com/en/donate/help/
Related PRs/issues: #11391 

**[Note]**: Please note that the redirect will not work yet, as we have to set this up in form assembly once the changes have been merged in. For now, I have added steps to confirm that UTM parameters are not lost in the "Steps to test" section.

This PR updates the donate help page to look for the query parameter `thank_you=true`. 

If this query parameter is found, the page renders a thank you message instead of the form assembly form. 

This is added so we can redirect users to the donate help page with this new "thank you" message after form submission, instead of redirecting to a form assembly hosted page with an un-styled "thank you" message.

This PR also adds tests to ensure that the UTM parameter is being added correctly with and without existing UTM parameters.

Lastly, this PR adds some updated instructions regarding the new field/changes.


# Screenshots

(Before) This is what the user **used** to see when they submitted the form and were redirected:

![Screenshot 2023-11-29 at 16-15-54 https __mozillafoundation tfaforms net](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/75350c51-6a7a-4d3b-899d-d2df283afcf0)


(After Update) Before form submission:
![Screenshot 2023-11-29 at 16-03-34 Donor Help and FAQ](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/42a92f27-9707-4daf-bc71-217053606787)


(After Update) After form submission and redirect:
![Screenshot 2023-11-29 at 16-05-23 Donor Help and FAQ](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/c8ebf38c-50d5-498f-add6-55db1c89b0a4)


# Steps to test
1. Visit https://foundation-s-11391-dona-arwuyu.herokuapp.com/en/donate/help/, and note that the form renders as expected.
2. Open the dev inspector, and in the search bar type "tfa_236". Note that highlighted element has a value of `<URL>/donate/help/?thank_you=true`.
<img width="1918" alt="Screenshot 2023-11-29 at 6 43 04 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/b61c08e5-81ad-4f37-bf6c-bf96fc748502">


3. Visit https://foundation-s-11391-dona-arwuyu.herokuapp.com/en/donate/help/?hello=there, and note that the form renders as expected.
4. Open the dev inspector, and in the search bar type "tfa_236". Note that highlighted element has a value of `<URL>/donate/help/?hello=there&thank_you=true`.
<img width="1918" alt="Screenshot 2023-11-29 at 6 43 34 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/9b7473cf-eab9-4504-8e82-ea436af6352d">

5. Visit https://foundation-s-11391-dona-arwuyu.herokuapp.com/en/donate/help/?thank_you=true, and note that the thank you message appears in place of the form.

![Screenshot 2023-11-29 at 18-51-01 Donate Help](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/28a09b8e-c829-4335-9a5b-3f9a5f9859ff)

6. If everything is working as expected, testing is complete!
